### PR TITLE
 I18n - Allow custom translation loaders to be injected via module config #6244

### PR DESCRIPTION
**Pull request by rodmcnew**
This change allows modules to add their own translation loaders to the I18n translator via config like this:

``` php
'translator_loaders' => array(
    'factories' => array(
        'my_new_loader' => 'ZendI18nDoctrineLoader\Factory\LoaderFactory',
    )
),
```

Before there was no way to add translation loaders without defining your own new translator service. See the ticket for more details: https://github.com/zendframework/zf2/issues/6244

### CODE REVIEW

File: library/Zend/ModuleManager/Feature/TranslatorPluginProviderInterface.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (samsonasik) [inline library/Zend/ModuleManager/Feature/TranslatorPluginProviderInterface.php]: 2015 ;)

File: library/Zend/Mvc/Service/TranslatorPluginManagerFactory.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (samsonasik) [inline library/Zend/Mvc/Service/TranslatorPluginManagerFactory.php]: 2015 ;)

**Comment by rodmcnew** [comment]
Closing and opening PR to trigger travis rebuild...

**Comment by rodmcnew** [comment]
The build failed in the following file which I don't believe has anything to do with this PR. Is there a way for me to trigger a Travis Rebuild?

/home/travis/build/zendframework/zf2/tests/ZendTest/Cache/Storage/Adapter/CommonAdapterTest.php:862

**Comment by Ocramius** [comment]
@rodmcnew that's (sadly) a test that fails every time the I/O is very slow in the travis infrastructure.

**Comment by rodmcnew** [comment]
I see, thanks @Ocramius.

File: library/Zend/Mvc/Service/ServiceListenerFactory.php [hunk 3]
```diff
@@ -59,6 +59,7 @@ class ServiceListenerFactory implements FactoryInterface
             'InputFilterManager'             => 'Zend\Mvc\Service\InputFilterManagerFactory',
             'LogProcessorManager'            => 'Zend\Mvc\Service\LogProcessorManagerFactory',
             'LogWriterManager'               => 'Zend\Mvc\Service\LogWriterManagerFactory',
+            'TranslatorPluginManager'        => 'Zend\Mvc\Service\TranslatorPluginManagerFactory',
```
Comment (weierophinney) [inline library/Zend/Mvc/Service/ServiceListenerFactory.php]: We try to keep these alphabetized; I can make that change on merge, however.

**Comment by weierophinney** [comment]
Merged for 2.4.0; @rodmcnew , please do a PR against https://github.com/zendframework/zf2-documentation with documentation of this feature. :)

**Pull request closed without merging**